src/library_syscall.js

```diff
@@ -65,7 +65,7 @@ var SyscallsLibrary = {
       return 0;
     },
     doMsync: function(addr, stream, len, flags) {
-      var buffer = new Uint8Array(HEAPU8.buffer, addr, len);
+      var buffer = new Uint8Array(HEAPU8.subarray(addr, addr + len));
```

Comment:
As mentioned above, `subarray` is fully supported and ok to use. `heapAndOffset` is a convenience for the specific case where we need not a `slice` but a `slice, offset`.